Bugfix/esp32 factory defaults

TD-er opened the pull request.

TD-er committed:
[HW CDC] Fix delay when no client connected to read serial logs

TD-er committed:
[UDP] Only process UDP packets while connected

TD-er committed:
[Factory Reset] Fix applying NVS stored settings only when these exist

TD-er committed:
[ESP-IDF5.1] Add custom_ESP32s2_4M316k_LittleFS_CDC

TD-er committed:
[Factory Reset] Add default flags for factory reset

TD-er committed:
[ESP-IDF5.1] Fix starting DHCP on ESP AP during setup

Fixes: #4931

TD-er committed:
[Cleanup] Reduce bin size (NVS storage & P128 NeoPixelBus)

TD-er committed:
[Cleanup] Reduce build size (P128 NeoPixelBus)

TD-er committed:
[Cleanup] Reduce build size (P104/P128)

TD-er committed:
[Cleanup] Reduce build size (P104 load/save functions)

TD-er committed:
[Cleanup] Reduce build size

TD-er committed:
[Cleanup] Reduce build size (AdaFruitGFX_Helper fonts)

TD-er committed:
[ESP32] Disable PSRAM for ESP32 4M builds (WT32-ETH01 issues)

TD-er committed:
[Docs] Add credits to origin of ESP32-S3 flash/PSRAM options

TD-er committed:
[ESP-IDF5.1] Improve IPv6 for ESP32-xx

tonhuisman commented on platformio_esp32_solo1.ini:
```diff
@@ -18,7 +18,7 @@ build_unflags             = ${esp32_base.build_unflags}
 [esp32_solo1_common_LittleFS]
 extends                   = esp32_base_idf5
-platform_packages         = framework-arduino-solo1 @ https://github.com/Jason2866/esp32-arduino-lib-builder/releases/download/1895/framework-arduinoespressif32-solo1-release_v5.1-8f01299.zip
+platform_packages         = framework-arduino-solo1 @ https://github.com/Jason2866/esp32-arduino-lib-builder/releases/download/1915/framework-arduinoespressif32-solo1-release_v5.1-d4d5f8a.zip
```
Is there a difference with /1916/ in the `core_defs` file?

TD-er replied:
Not sure what you mean?

TD-er committed:
[ESP-IDF5.1] Fix IPv6 for ESP32-S3

TD-er committed:
[Build] Fix build due to typo in source code

TD-er committed:
Fix "Enable SDK WiFi Auto Reconnect" state on sysinfo page (#4935)

Fixes: #4935

TD-er committed:
[build] Fix another build issue due to includes

TD-er committed:
Revert "[Cleanup] Reduce build size (AdaFruitGFX_Helper fonts)"

This reverts commit a95ad3bda183f0ecf4e41efff3e9cbe4fb3a2e16.

Ton was already working on this in a separate PR.
Sorry Ton I forgot to check it...

TD-er committed:
[Factory reset] Fix using factory reset defaults on new devices

TD-er committed:
[ESP-IDF5.1] Update to latest ESP-IDF & Arduino fixes

TD-er committed:
Fix crash in command arg parsing with nullptr string

TD-er committed:
[IPv6] Allow `ip6` command from all command sources

TD-er committed:
[P077] Add `cseclearpulses` command to reset CF pulses counter

As requested on the [forum](https://www.letscontrolit.com/forum/viewtopic.php?t=10006)

TD-er committed:
[Cleanup] Reduce build size by simplifying logs

TD-er committed:
[Provisioning] Show error and event on failed firmware update (#4941)

Fixes: #4941

TD-er committed:
[Build] Fix build error on ESP8266

TD-er committed:
Merge remote-tracking branch 'letscontrolit/mega' into bugfix/ESP32_factoryDefaults
